mcpserver/tools.go

```diff
@@ -479,19 +467,21 @@ Failure to do so will result in the tool being stuck, awaiting for the command t
 		),
 	),
 	Handler: func(ctx context.Context, request mcp.CallToolRequest) (*mcp.CallToolResult, error) {
-		envID, err := request.RequireString("environment_id")
+		repo, env, err := openEnvironment(ctx, request)
 		if err != nil {
-			return nil, err
-		}
-		env := environment.Get(envID)
-		if env == nil {
-			return mcp.NewToolResultError(fmt.Sprintf("environment %s not found", envID)), nil
+			return mcp.NewToolResultErrorFromErr("unable to open the environment", err), nil
 		}
+
 		command := request.GetString("command", "")
 		shell := request.GetString("shell", "sh")
 
 		background := request.GetBool("background", false)
 		if background {
+			// We want to update the repository whether the command fails or not
+			defer func() {
+				_ = repo.Update(ctx, env, "Run "+command, request.GetString("explanation", ""))
```

Comment:
update calls definitely need error handling unless i'm missing something...

Reply:
Workaround to make git notes propagate on command failure, e.g. calling it whether the command works or not.

I agree this is too naive

Reply:
we at least gotta be logging errors from this, if not populating rerr with some sort of wrapped error on failure

Reply:
Fixed -- PTAL